flow-decomposition/src/main/java/com/powsybl/flow_decomposition/FlowDecompositionComputer.java

```diff
@@ -124,20 +138,27 @@ private void decomposeFlowForState(Network network,
                                        LoadFlowRunningService.Result loadFlowServiceAcResult) {
         saveAcReferenceFlow(flowDecompositionResultsBuilder, xnecList, loadFlowServiceAcResult);
         compensateLosses(network);
+        observers.computedAcFlows(network, loadFlowServiceAcResult);
```

Comment:
Could you refactor this with the saveAcReferenceFlow method to avoid duplicating the flow extraction code ?
I don't think the observer really need to have the AC flows on each branch of the network (even from non-monitored lines (XNEC))
You can do the same thing for DC flow and nodal injections

Reply:
I am not sure I understand this comment, you seem to have 2 different points: some code is duplicated and some observer methods are useless.

About the usefulness of having AC flows on the whole network, our client needed to have them to validate the methodology, they couldn't only check the results from the XNEC flows.

About the duplication, I think I see what could be done. I'll try something

Reply:
@OpenSuze I made some slight changes in the last commit.

I don't see how I can refactor more than that to make it simpler. If you had something in mind, I would be happy if you would share it :)

Reply:
If your client want to have the AC flow on each branch, then your implementation is ok for me ! 
I had something in mind if you only required the AC flow on XNECs.
Thank you for the refactoring nonetheless !